File: JAVA/GroupAnagrams.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class GroupAnagrams {

    public static void main(String[] args) {
        System.out.printf("Resultado: %d\n", groupAnagrams( new String[]{"a"} ));
    }

    public static List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> hashmap = new HashMap<>();

        for (String str : strs) {
            char[] tmp = str.toCharArray();
            Arrays.sort(tmp);
            String sortStr = new String(tmp);
            hashmap.computeIfAbsent(sortStr, k -> new ArrayList<>()).add(str);
        }

        return new ArrayList<>(hashmap.values());
    }

}

File: GO/GroupAnagrams.go
/**
 * @version GO 1.21.6
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import "sort"

func groupAnagrams(strs []string) [][]string {
	hashmap := map[string][]string{}
	for _, str := range strs {
		tmp := []byte(str)
		sort.Slice(tmp, func(i, j int) bool {
			return tmp[i] < tmp[j]
		})

		sortStr := string(tmp)
		hashmap[sortStr] = append(hashmap[sortStr], str)
	}
	result := make([][]string, 0, len(hashmap))
	for _, v := range hashmap {
		result = append(result, v)
	}

	return result
}
